Add webcompat-release-hotfix XPI

denschub opened the pull request:
We sometimes (for example right now) need to create an .xpi that only contains specific new interventions for rollout to release users. We usually do not want to roll out the main branch to release, as that may contain additional code changes that would significantly increase our risk, and make testing a lot more complicated.

escapewindow commented:
There are questions here about how best to do a dot release, when master is 2 versions ahead of mozilla-release. This works, but may require force-push, which prevents using branch protections.

We could:
- Skip branch protections, and do extra reviews when signing off? How would the person signing off know we need to check git history?
- Enable branch protections, merge in master from the point we would have branched off of, then apply the dot release revision(s), and review that as a PR to the release hotfix branch
- Create a new release branch every time we want to release a dot release. This either means we update the xpi manifest every time, or we somehow update shipit and the xpi signing logic to be able to take wildcards/regexes/?

There are security implications to getting rid of the branch protections, and workflow/process implications otherwise.

escapewindow reviewed (changes requested):
Marking `request changes` so we know we need to address https://github.com/mozilla-extensions/xpi-manifest/pull/93#issuecomment-747620120 before merging :)

denschub commented:
@escapewindow I have enabled branch protections for our `release-hotfix` now. We're fine with the little bit of extra process on our end, and the possibility for a diverged release branch (which will, in reality, probably have some extra merge-commits in the future). :)

I also rebased the change so that instead of adding the branch to the now removed `xpi-manifest.yml`, I created a new file in the `manifests` folder. Let me know if that works for you!

escapewindow reviewed (approved):
Thank you!
